collector/kernel/bpf_handler.cc

```diff
@@ -320,3 +324,8 @@ void BPFHandler::check_cb(std::string error_loc)
     log_.warn("after {}, cumulative lost count non-zero: {} new, {} total", error_loc, diff_lost_count, lost_count);
   }
 }
+
+void BPFHandler::debug_bpf_lost_samples()
```

Comment:
I consider wrapping this and the other related debug code in #ifndef NDEBUG, but left it in for release builds also following the same pattern as other debug code and because it is also useful to run it against release code.  If someone has a preference for having it in debug builds only please let me know.

Reply:
I think there's two types of debug code to consider:

1. Debug data collection - this may be internal metrics that could be of use to extract when something goes wrong.  This can be then broken out into 'often useful vs. not useful,' 'low vs. high overhead' and 'anonymous vs. customer data'.  The ones that are normally safe to 'keep around' in production are those that fall into the first part of all of these categories.
2. Side effects - these are code or other operations that change behavior from the normal, expected 'normal' operating mode.  These should likely never end up in production, as they can become fodder for obscure bugs, DoS and otherwise increase the coverage tree and exposure suface.

I think, in this case, the above function falls firmly into the latter category - and if there's similar functionality floating around, we likely should make sure we get around to addressing it.

Reply:
The other, similar, debug code I referred to is:
      --crash                           internal development - force a SIGSEGV
      --schedule-crash=[schedule-crash] internal development - will force a
                                        SIGSEGV after given number of seconds
(there may be others)

I'll propose that 1) I change the new debug code to inject lost bpf samples to be in debug builds only, 2) I'll do the same with the above debug code, and 3) that in the future we consider having other build type(s) in addition to debug (debug logs and code enabled, debug symbols present, possibly lower optimization levels, etc.) and release, maybe DebugRelease that includes some debug code/logs in what is otherwise a release build.

Reply:
Works for me.  Thank you very much.

Reply:
Done.